src/com/palmergames/bukkit/towny/command/NationCommand.java

```diff
@@ -1847,4 +1858,206 @@ public void nationRename(Player player, Nation nation, String newName) {
+                    return;
+                }
+
+                nation = resident.getTown().getNation();
+                notAffordMSG = TownySettings.getLangString("msg_err_cant_afford_tp");
+
+                nationSpawn(player, split, nation, notAffordMSG);
+
+            } else {
+                // split.length > 1
+                nation = TownyUniverse.getDataSource().getNation(split[0]);
+                notAffordMSG = String.format(TownySettings.getLangString("msg_err_cant_afford_tp_nation"), nation.getName());
+
+                nationSpawn(player, split, nation, notAffordMSG);
+
+            }
+        } catch (NotRegisteredException e) {
+
+            throw new TownyException(String.format(TownySettings.getLangString("msg_err_not_registered_1"), split[0]));
+
+        }
+
+    }
+
+    /**
+     * Core nation spawn function to allow admin use.
+     *
+     * @param player
+     * @param split
+     * @param nation
+     * @param notAffordMSG
+     */
+    public static void nationSpawn(Player player, String[] split, Nation nation, String notAffordMSG) {
+
+        try {
+            boolean isTownyAdmin = TownyUniverse.getPermissionSource().has(player, PermissionNodes.TOWNY_COMMAND_TOWNYADMIN_NATION_SPAWN_OTHER.getNode());
+            Resident resident = TownyUniverse.getDataSource().getResident(player.getName());
+            Location spawnLoc;
+            TownSpawnLevel nationSpawnPermission;
+
+            spawnLoc = nation.getNationSpawn();
+
+            // Determine conditions
+            if (isTownyAdmin) {
+                nationSpawnPermission = TownSpawnLevel.ADMIN;
+            } else if ((split.length == 0)) {
+				nationSpawnPermission = TownSpawnLevel.PART_OF_NATION;
+            } else {
+                // split.length > 1
+                if (!resident.hasTown()) {
+                    nationSpawnPermission = TownSpawnLevel.UNAFFILIATED;
+                }
+                else if (resident.hasNation()) {
+                    Nation playerNation = resident.getTown().getNation();
+                    Nation targetNation = nation;
+
+                    if (playerNation == targetNation) {
+                        nationSpawnPermission = TownSpawnLevel.PART_OF_NATION;
+                    } else if (targetNation.hasEnemy(playerNation)) {
+                        // Prevent enemies from using spawn travel.
+                        throw new TownyException(TownySettings.getLangString("msg_err_public_spawn_enemy"));
+                    } else if (targetNation.hasAlly(playerNation)) {
+                        nationSpawnPermission = TownSpawnLevel.NATION_ALLY;
+                    } else {
+                        nationSpawnPermission = TownSpawnLevel.UNAFFILIATED;
+                    }
+                } else {
+                    nationSpawnPermission = TownSpawnLevel.UNAFFILIATED;
+                }
+            }
+
+            nationSpawnPermission.checkIfAllowed(plugin, player, resident.getTown());

```

Comment:
Could you also check this line?
I am not too sure if this is working as intended. Not too familiar with townies permission handling.
Maybe I should just put in a check manually my self instead of using the TownSpawnLevel for this.

Reply:
This you can leave as it is for now.